docs/decisions/0019-scaling-styles-with-design-tokens.rst

```diff
@@ -0,0 +1,261 @@
+
+Implications for theme authors (e.g., ``@edx/brand`` packages)
+------------------------------------------------------------
+
+Currently, Paragon recommends theme authors to create a theme package such as ``@edx/brand-openedx`` (`Github <https://github.com/openedx/brand-openedx>`__) and ``@edx/brand-edx.org`` (`Github <https://github.com/edx/brand-edx.org>`__).
+
+While the migration from SCSS variables to CSS variables is a breaking change for theme authors, we have tried to mitigate this by keeping the existing SCSS variables but defining them such that their values refer to the new CSS variables. Because SCSS can't evaluate the CSS variable at runtime, it utilizes the CSS variable in the resulting output CSS used in the browser.
+
+Currently, theme authors largely modify SCSS variables from core Paragon by creating a ``_variables.scss`` file and importing it *after* the core Paragon SCSS styles in consuming applications (e.g., micro-frontends). Doing so, SCSS will override the original variables' values defined by core Paragon with the new SCSS from the ``@edx/brand`` theme.
+
+With design tokens, theme authors will instead override core Paragon tokens by defining their own JSON tokens that get deep merged alongside the core Paragon tokens, thus overriding any tokens that were defined by the theme author.
+
+This approach gives theme authors the same theming experience as core Paragon's tokens architecture though theme authors could also hardcode the CSS variables themselves like they do with SCSS variables today. The former approach is primarily recommended by Paragon to enable theme authors to have same theming approach as core Paragon.
+
+Furthermore, given CSS variables may be generated and consumed by applications in numerous ways, it's worth mentioning alternative strategies as well.
+
+For example, the above approach largely assumes you're able to generate and use a CSS file containing CSS variable overrides in consuming applications for each desired theme. It may be impractical to generate and host a CSS file for each dynamically generated theme, e.g. if the theme is driven by user input or the theme values (like HEX values) are stored in a database, retrieved by an API.
+
+In such cases, consumers could also directly override CSS variables at runtime by generating and injecting them into the ``<head>`` of the HTML document (e.g., with ``react-helmet``) after retrieving the values from an API, which would result in overriding the core Paragon CSS variables as well.
+
+Future considerations: Customizing the theme via a user interface
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+More forward thinking, we would also like to explore creating a UI on top of these design tokens such that the understanding and writing of JSON files is abstracted away from theme authors. Such a UI may also enable designers to self-serve update the theme.
```

Comment:
[naive question] Is there a way for Figma to be a potential source of truth for the design tokens via some sort of export?  Just thinking about designers getting to use a powerful industry standard tool they're familiar with vs. a tool we have to maintain.

Reply:
Yep, this is definitely part of the long term vision! For example, the [Tokens Studio](https://tokens.studio/) plugin for Figma has a GitHub Sync feature that could enable a two-way sync between design tokens defined in code and Figma, where designers could even self-serve update the value of design tokens and see their changes propagate throughout the whole system with minimal engineering effort (e.g., changing a single color across the platform).

I'm surprised this wasn't included in the example potential benefits here! I will include some words about this.